natter.py

```diff
@@ -1328,7 +1328,7 @@ def _discover(self):
             while True:
                 try:
                     buff, addr = sock.recvfrom(4096)
-                    m = re.search(r"LOCATION: *(http://[^\[]\S+)\s+",
+                    m = re.search(r"(?:LOCATION|Location): *(http://[^\[]\S+)\s+",
```

Comment:
To get compatible with [MicroPython's re](https://docs.micropython.org/en/latest/library/re.html), we should avoid the use of non-capturing groups `(?:...)` as it is not supported.

The following change is suggested:

```suggestion
                    m = re.search(r"[Ll][Oo][Cc][Aa][Tt][Ii][Oo][Nn]: "
                                  r"*([Hh][Tt][Tt][Pp]://[^\[]\S+)\s+",
```